packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const error = require('../error');
+const isA = require('joi');
+const ScopeSet = require('fxa-shared').oauth.scopes;
+const validators = require('./validators');
+
+module.exports = (log, db, config, customs, push, oauthdb, subhub) => {
+  const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
+    'https://identity.mozilla.com/apps/subscriptions';
+
+  async function handleAuth(auth) {
+    const scope = ScopeSet.fromArray(auth.credentials.scope);
+    if (! scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
```

Comment:
@rfk can clarify here too, we may want to differentiate between read and write access. URL scopes grant [read-write access](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/fxa-oauth-server/docs/scopes.md#url-scopes) by default, though some of these endpoints seem like they only require read-access. If we decide on differentiating between the two, that can be in a follow-on PR.

Reply:
Yeah, I think this is what I filed #882 for. If we're not using `profile:email` as the scope for reading subscription status, then we'll need a different one. We definitely don't want any client other than the management pages to have access to write APIs for subscription stuff.

Reply:
This seems reasonable to me to start with; [this doc](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/fxa-oauth-server/docs/scopes.md) has some suggested notation for distinguishing read-vs-write scopes, bit I tink having completely separate scopes for "view the user's subscription as part of their profile data" and "manage the user's subscription state" makes sense.  IIUC you would *never* expect an RP to be calling the `/subscription` APIs directly, even in read-only mode, right?

Reply:
> IIUC you would never expect an RP to be calling the /subscription APIs directly, even in read-only mode, right?

Yes, the only subscription-related APIs used by RPs live on the profile server, to fetch the user's set of relevant subscription capabilities. These subscription management APIs are only for use by our own subscription management pages.

> When requesting a grant for SUBSCRIPTIONS_MANAGEMENT_SCOPE we should ensure the client_id making the request is actually allowed to request the grant.

This should be a part of #881